Clinical-Genomics/microSALT
# Added support to run ANY resistance DB

# Description

_The features of this PR primarily concerns end-users_

_Summary of the changes made:_

- [x] Removed explicit ExPEC integration
- [x] Added cli line to add custom virulence calls
- [x] Added VFDB parsing support
- [x] Put if case around swedac logotype based on RUNID
- [x] Add MultiQC run to microSALT
- [x] Finish up microSALT webserver support
- [x] Fixed a bug where a new allele was reported as a known ST
- [x] Added CLI command to update databases
- [x] Add custom_target to former ExPEC entries.

## Primary function of PR

# Testing
_If the update is a hotfix, it is sufficient to rely on the development testing along with the Travis self-test automatically applied to the PR._

_Test routine to verify the stability of the PR:_


- _`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-microsalt-stage.sh BRANCHNAME`_
- _`us`_
- _`source activate S_microSALT`_


- Start analysis of ACC7470 normally
- Start analysis of ACC5551 with --custom_target flag pointing to https://github.com/Clinical-Genomics/microSALT/tree/master/unique_references/ExPEC.fsa
- Verify that the results for projects ACC5551 are consistent with the results attached to AMSystem doc 1490, Microbial_WGS.xlsx
  - Check that ACC7470 doesn't have the SWEDAC logo
  - Check that ACC7470 has a New allele for it's length-deprived sample
  - Look for MultiQC results in e-mailed results, make sure the report isnt broken
- Run microSALT utils refer update. Make sure it works / doesn't error and crash.
- Run microSALT utils view. Make sure the started webpage / webserver looks ok.


## Test results
_These are the results of the tests, and necessary conclusions, that prove the stability of the PR._

# Sign-offs
- [ ] Approved to run at Clinical-Genomics by @sylvinite
